cpp/src/arrow/io/compressed.cc

```diff
@@ -413,7 +413,7 @@ class CompressedInputStream::Impl {
     RETURN_NOT_OK(buf->Resize(bytes_read));
     // Using std::move because some compiler might has issue below:
     // https://wg21.cmeerw.net/cwg/issue1579
```

Comment:
Hmm. We may not be able to remove this `std::move()`: https://github.com/apache/arrow/pull/39807

Let's try R jobs.